code/game/machinery/iv_drip.dm

```diff
@@ -1,224 +1,684 @@
+				if(91 to INFINITY)	filling.icon_state = "[tipped ? "tipped_" : ""]reagent100"
 
 			var/reagent_color = reagents.get_color()
 			filling.icon += reagent_color
 			add_overlay(filling)
-
-/obj/machinery/iv_drip/MouseDrop(over_object, src_location, over_location)
-	..()
-
-	if(attached)
-		visible_message("[src.attached] is detached from \the [src].")
-		src.attached = null
-		src.update_icon()
-		blood_message_sent = FALSE
+		if(attached)
+			add_overlay("iv_in[tipped ? "_tipped" : ""]")
+			if(mode)
+				add_overlay("light_green[tipped ? "_tipped" : ""]")
+			else
+				add_overlay("light_red[tipped ? "_tipped" : ""]")
+		else
+			add_overlay("iv_out[tipped ? "_tipped" : ""]")
+	if(tank)
+		if(istype(tank, /obj/item/tank/oxygen))
+			tank_type = "oxy"
+		else if(istype(tank, /obj/item/tank/anesthetic))
+			tank_type = "anest"
+		else if(istype(tank, /obj/item/tank/phoron))
+			tank_type = "phoron"
+		else
+			tank_type = "other"
+		add_overlay("tank_[tank_type][tipped ? "_tipped" : ""]")
+		update_gauge()
+	if(breath_mask)
+		if(breather)
+			add_overlay("mask_on[tipped ? "_tipped" : ""]")
+		else
+			add_overlay("mask_off[tipped ? "_tipped" : ""]")
+		if(epp_active)
+			add_overlay("light_blue[tipped ? "_tipped" : ""]")
+
+/obj/machinery/iv_drip/proc/update_gauge()
+	var/gauge_pressure = 0
+	var/last_gauge_pressure
+	if(tank.air_contents)
+		gauge_pressure = tank.air_contents.return_pressure()
+		if(gauge_pressure > TANK_IDEAL_PRESSURE)
+			gauge_pressure = -1
+		else
+			gauge_pressure = round((gauge_pressure/TANK_IDEAL_PRESSURE)*tank.gauge_cap)
+	if(gauge_pressure == last_gauge_pressure)
 		return
+	last_gauge_pressure = gauge_pressure
+	add_overlay("[tank.gauge_icon][(gauge_pressure == -1) ? "overload" : gauge_pressure][tipped ? "_tipped" : ""]")
 
-	if(in_range(src, usr) && ishuman(over_object) && get_dist(over_object, src) <= 1)
-		visible_message("[usr] attaches \the [src] to \the [over_object].")
-		src.attached = over_object
-		src.update_icon()
-
+/obj/machinery/iv_drip/machinery_process()
+	if(istype(breather)) // Mask secured
+		if(!breather.Adjacent(src))
+			step_to(src, get_turf(breather), 1)
+			if(world.time > last_creak + 10 SECONDS)
+				last_creak = world.time
+				src.visible_message("\The [src]'s wheels creak as it slowly gets tugged towards [breather] by \the [breath_mask]'s cable.")
+				playsound(src, 'sound/effects/roll.ogg', 50, 1)
+				shake_animation(2)
+			if(get_dist(src, breather) >= 2)
+				src.visible_message("\The [src] jerks as \the [breath_mask]'s cable is pulled taut!", SPAN_WARNING("You feel \the [src] jerk as your [breath_mask]'s cable is pulled taut."))
+				shake_animation(4)
+				if(prob(40))
+					do_crash()
+
+		if(valve_open)
+			if(breather.species.flags & NO_BREATHE)
+				src.visible_message(SPAN_NOTICE("\The [src] buzzes, automatically deactivating \the [tank]."))
+				playsound(src, 'sound/machines/buzz-two.ogg', 50)
+				tank_off()
+				update_icon()
+				return
```

Comment:
Some of the returns in the breather processing logic will prevent the IV drip from processing, when it should. Having the mask deactivate shouldn't skip the IV drip for that tick.

Reply:
Yeah, I realized a bit late regarding how some of the previous checks would do that for the injecting/intaking mode for the IV when someone was wearing the mask and changed it accordingly, mostly by moving the checks and returns to only happen if the valve is open. The returns are still there to make sure it won't continue to proceed and be fully certain it'll prevent any strange behavior. The current iteration of the checks/returns is the best I managed to get to

Reply:
Same here. I can help correct this if you want, just have to ask (same for any of my review comments).

Reply:
I really don't know if that's even possible without gutting out the entire code for the Breather/Attached again, or adding yet another check to just jump lines (if that's even a thing), or some ``proc/attached_in()`` nonsense that'd make this even more confusing

Reply:
You can do something like this so the returns only return from the breather OR attached portions rather than the entire machinery proc.

[Example Code For You](https://github.com/mikomyazaki/Aurora.3/blob/3784724babe29f3f5949439fcdab25fca7744a81/code/game/machinery/iv_drip.dm#L169)

Reply:
So the answer really was the fucking nonsense, oh my god

Reply:
Is it nonsense? You could also make it work by rearranging the ifs + using more elses. But I think that gets more tangled.

Reply:
Nonsense because at the time I considered it, I was already stressed and began to over-complicate the possible solution into something I thought as unreasonable. Though now I'll definitely apply that

Reply:
And done! Thank you very much!